### PR TITLE
feat: add release check dashboard

### PR DESCRIPTION
**Pull request by SebastianBezold**
## Description

This PR adds a new directory `trg-checks-dashboard` that contains a Golang CLI app. This app is running our automated TRG checks and compiles the results grouped by product to a dashboard. 
There is also a workflow, that can be run on demand to trigger the dashboard generation. It requires a GitHub token as input to authenticate the API requests with. The dashboard will then be deployed as GitHub pages

## Additional info

Requires GitHub pages to be activated for sig-release before the workflow does successfully run. 
See eclipse-tractusx/.eclipsefdn#20

Related to system team issue eclipse-tractusx/sig-infra#88

### CODE REVIEW

File: trg-checks-dashboard/internal/templating/tractusx.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	return checkedRepo
+}
+
+func initializeChecksForDirectory(dir string) []tractusx.QualityGuideline {
+	var checks []tractusx.QualityGuideline
+
+	checks = append(checks, docs.NewReadmeExists(dir))
+	checks = append(checks, docs.NewInstallExists(dir))
+	checks = append(checks, docs.NewChangelogExists(dir))
+	//checks = append(checks, repo.NewRepoStructureExists(dir))
```
Comment (tomaszbarwicki) [inline trg-checks-dashboard/internal/templating/tractusx.go:106]: Not sure why we may not want this check to happen, any reason why we repo structure check commented out?

File: trg-checks-dashboard/web/templates/index.html.tmpl [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
```
Comment (tomaszbarwicki) [inline trg-checks-dashboard/web/templates/index.html.tmpl:1]: Don't we need a copyright header for this one?

File: trg-checks-dashboard/cmd/build.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		ensureOutputDirExists()
+		copyAssets()
+
+		var outputBuffer bytes.Buffer
+		products, unhandledRepos := templating.CheckProducts()
+		// unhandledRepos := getDemoUnhandledRepos()
+
+		// templating.RenderHtmlTo(&outputBuffer, &templating.TemplateData{Config: getConfig(), CheckedProducts: getDemoChecks(), UnhandledRepos: unhandledRepos})
```
Comment (tomaszbarwicki) [inline trg-checks-dashboard/cmd/build.go:51]: Do we need these two if commented?

Comment (tomaszbarwicki) [inline trg-checks-dashboard/cmd/build.go:51]: Not sure if we need these two if commented? Not clear on the intention.

**Review by tomaszbarwicki — CHANGES_REQUESTED** [review]
Good work! Just little comments here and there. Loose suggestion that perhaps might be worth considering some basics tests.

File: trg-checks-dashboard/internal/templating/tractusx.go (hunk 1 above)
Comment (SebastianBezold) [reply trg-checks-dashboard/internal/templating/tractusx.go:106]: Oh I actually forgot to mentioned it in the PR description. There is an issue that I didn't see when adding the "baseDir" addition. eclipse-tractusx/tractusx-quality-checks#42 -> the call stack uses a function, that is not able to take a baseDir and fails. Therefor the whole check fails, when running on a different dir than the current one

File: trg-checks-dashboard/web/templates/index.html.tmpl (hunk 2 above)
Comment (SebastianBezold) [reply trg-checks-dashboard/web/templates/index.html.tmpl:19]: thanks for spotting. Added!

File: trg-checks-dashboard/cmd/build.go (hunk 3 above)
Comment (SebastianBezold) [reply trg-checks-dashboard/cmd/build.go]: yup indeed. just forgot to remove it. now gone!

**Review by tomaszbarwicki — APPROVED** [review]
LGTM

**Pull request merged**